tests/functional/s/shallow_copy_environ.py

```diff
@@ -27,3 +27,11 @@
 import os
 
 copy.deepcopy(os.environ)
+
+# Edge cases
+copy.copy()  # [no-value-for-parameter]
+copy.copy(x=test_dict)
+copy.copy(x=os.environ)  # [shallow-copy-environ]
+copy.copy(**{"x": os.environ})  # [shallow-copy-environ]
+copy.copy(**{"y": os.environ})  # [unexpected-keyword-arg]
+copy.copy(y=os.environ)  # [no-value-for-parameter, unexpected-keyword-arg]
```

Comment:
I would expect the last two lines to raise the same set of messages. We don't use inference to analyze whether **kwargs contains the missing named parameter `x`. We could.

Reply:
Captured in #8785